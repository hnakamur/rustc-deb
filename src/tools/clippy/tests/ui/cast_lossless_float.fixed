// run-rustfix

#[warn(clippy::cast_lossless)]
#[allow(clippy::no_effect, clippy::unnecessary_operation)]
fn main() {
    // Test clippy::cast_lossless with casts to floating-point types
    f32::from(1i8);
    f64::from(1i8);
    f32::from(1u8);
    f64::from(1u8);
    f32::from(1i16);
    f64::from(1i16);
    f32::from(1u16);
    f64::from(1u16);
    f64::from(1i32);
    f64::from(1u32);
}
