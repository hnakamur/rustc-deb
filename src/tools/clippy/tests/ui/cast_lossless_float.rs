// run-rustfix

#[warn(clippy::cast_lossless)]
#[allow(clippy::no_effect, clippy::unnecessary_operation)]
fn main() {
    // Test clippy::cast_lossless with casts to floating-point types
    1i8 as f32;
    1i8 as f64;
    1u8 as f32;
    1u8 as f64;
    1i16 as f32;
    1i16 as f64;
    1u16 as f32;
    1u16 as f64;
    1i32 as f64;
    1u32 as f64;
}
