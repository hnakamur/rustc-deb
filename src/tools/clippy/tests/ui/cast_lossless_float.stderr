error: casting i8 to f32 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:7:5
   |
LL |     1i8 as f32;
   |     ^^^^^^^^^^ help: try: `f32::from(1i8)`
   |
   = note: `-D clippy::cast-lossless` implied by `-D warnings`

error: casting i8 to f64 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:8:5
   |
LL |     1i8 as f64;
   |     ^^^^^^^^^^ help: try: `f64::from(1i8)`

error: casting u8 to f32 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:9:5
   |
LL |     1u8 as f32;
   |     ^^^^^^^^^^ help: try: `f32::from(1u8)`

error: casting u8 to f64 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:10:5
   |
LL |     1u8 as f64;
   |     ^^^^^^^^^^ help: try: `f64::from(1u8)`

error: casting i16 to f32 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:11:5
   |
LL |     1i16 as f32;
   |     ^^^^^^^^^^^ help: try: `f32::from(1i16)`

error: casting i16 to f64 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:12:5
   |
LL |     1i16 as f64;
   |     ^^^^^^^^^^^ help: try: `f64::from(1i16)`

error: casting u16 to f32 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:13:5
   |
LL |     1u16 as f32;
   |     ^^^^^^^^^^^ help: try: `f32::from(1u16)`

error: casting u16 to f64 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:14:5
   |
LL |     1u16 as f64;
   |     ^^^^^^^^^^^ help: try: `f64::from(1u16)`

error: casting i32 to f64 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:15:5
   |
LL |     1i32 as f64;
   |     ^^^^^^^^^^^ help: try: `f64::from(1i32)`

error: casting u32 to f64 may become silently lossy if types change
  --> $DIR/cast_lossless_float.rs:16:5
   |
LL |     1u32 as f64;
   |     ^^^^^^^^^^^ help: try: `f64::from(1u32)`

error: aborting due to 10 previous errors

